testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -162,6 +162,22 @@ def test_AnalysisFromFunction(u, start, stop, step, nframes):
         assert_equal(results, ana.results)
 
 
+def test_AnalysisFromFunction_args_order(u):
+    # Issue 2503
+    def mass_xyz(atomgroup, masses):
+        return atomgroup.positions * masses
+    
+    protein = u.select_atoms('protein')
+    ans1 = base.AnalysisFromFunction(mass_xyz, u.trajectory, protein, protein.masses.reshape(-1, 1))
+    ans2 = base.AnalysisFromFunction(mass_xyz, protein, protein.masses.reshape(-1, 1))
+    
+    assert_equal(isinstance(ans1.args[0], mda.core.groups.AtomGroup), True)
+    assert_equal(isinstance(ans1.args[1], np.ndarray), True)
+    assert_equal(ans1.args[0], ans2.args[0])
+    assert_equal(ans1.args[1], ans2.args[1])
+    assert_equal(ans1.run().results, ans2.run().results)
```

Comment:
Here you could have used ``assert isinstance(ans1.args[0], mda.core.groups.AtomGroup)`` instead.

In general I think the test should be a bit stronger, and test that `args[0] is protein` and same for the masses. It should also check that there is no mysterious `args[2]`. It would also be best to test that `ans2._trajectory is protein.universe.trajectory`, since that is what the code is actually looking for.

After that you probably don't need to actually `run()` the analysis.